KomaMRIBase/test/runtests.jl

```diff
@@ -538,24 +538,24 @@ end
         dx = rand(Ns, Nt)
         dy = rand(Ns, Nt)
         dz = rand(Ns, Nt)
-        arbitrarymotion = @suppress ArbitraryMotion(t_start, t_end, dx, dy, dz)
-        t = times(arbitrarymotion)
+        arbitrarymotion = MotionVector(Trajectory(TimeRange(t_start, t_end), dx, dy, dz))
+        t = range(t_start, t_end, Nt)
         xt, yt, zt = get_spin_coords(arbitrarymotion, ph.x, ph.y, ph.z, t')
         @test xt == ph.x .+ dx
         @test yt == ph.y .+ dy
         @test zt == ph.z .+ dz
     end
 
-    simplemotion = SimpleMotion(
-        PeriodicTranslation(dx=0.05, dy=0.05, dz=0.0, period=0.5, asymmetry=0.5),
-        Rotation(pitch=0.0, roll=0.0, yaw=π / 2, t_start=0.05, t_end=0.5),
+    simplemotion = MotionVector(
+        Translation(times=Periodic(period=0.5, asymmetry=0.5), dx=0.05, dy=0.05, dz=0.0),
+        Rotation(times=TimeRange(t_start=0.05, t_end=0.5), pitch=0.0, roll=0.0, yaw=π / 2)
```

Comment:
> Motion to AbstractMotion (what methods need to be defined to create a new one? Docs)

I cannot do this since AbstractMotion is already defined as an abstract type:
```julia
struct MotionVector{T<:Real} <: AbstractMotion{T}
    motions::Vector{<:Motion{T}}
end
```
```julia
struct NoMotion{T<:Real} <: AbstractMotion{T} end
```
I will explain you why I needed to define this abstract type.
Ok with the rest of suggested changes

Reply:
There are two options

(1) rename the current `AbstractMotion` to something else, as it is not meant to be extended and the name makes more sense for `Translation`, `Rotation`, `Flow`, etc.

(2) remove the reason for its existence, like making `NoMotion{T<:Real}` an alias for an empty `MotionList` so we can dispatch based on it.

but I could be missing something.

Reply:
I have opted for the first option:
```julia
abstract type AbstractMotionList{T<:Real} end
```

Motion:
```julia
abstract type AbstractMotion{T<:Real} end

struct MotionList{T<:Real} <: AbstractMotionList{T}
    motions::Vector{<:AbstractMotion{T}}
end
```

NoMotion:
```julia
struct NoMotion{T<:Real} <: AbstractMotionList{T} end
```